veps/sig-compute/pcihole64-proposal.md

```diff
@@ -0,0 +1,323 @@
+`pcihole64` node below the `pcie-root` controller node in a domain XML.
+Setting the value of `pcihole64` to `0` disables the 64-Bit PCI hole.
+
+```xml
+<controller type='pci' index='0' model='pcie-root'>
+    <pcihole64 unit='KiB'>0</pcihole64>
+</controller>
+```
+
+When the feature is enabled, `virtwrap/converter` used by `virt-launcher` should
+configure the `pcihole64` on the `pcie-root` controller of an `api.Domain`
+object. The required configuration can be added as part of the conversion in
+`converter.Convert_v1_VirtualMachineInstance_To_api_Domain`.
+
+Since `converter.Convert_v1_VirtualMachineInstance_To_api_Domain` does not yet
+create `pcie-root` controllers, it must also be extended to add a basic
+`pcie-root` controller to `domain.Spec.Devices.Controllers` before configuring
+the `pcihole64` on it. This can be done before or after adding other controllers
+such as usb or scsi.
+
+To allow configuring the `pcihole64` on a `pcie-root` controller the feature
+requires an addition to `virtwrap/api/schema.go`.
+
+```golang
+type Controller struct {
+Type      string            `xml:"type,attr"`
+Index     string            `xml:"index,attr"`
+Model     string            `xml:"model,attr,omitempty"`
+Driver    *ControllerDriver `xml:"driver,omitempty"`
+Alias     *Alias            `xml:"alias,omitempty"`
+Address   *Address          `xml:"address,omitempty"`
+PCIHole64 *PCIHole64        `xml:"pcihole64,omitempty"`
+}
+
+type PCIHole64 struct {
+Value uint   `xml:",chardata"`
+Unit  string `xml:"unit,attr,omitempty"`
+}
+```
+
+## API Examples
+
+<!--
+Tangible API examples used for discussion
+-->
+
+By setting the `kubevirt.io/legacyWindowsCompat` annotation on a
+`VirtualMachine` object to `"true"` the feature is enabled. Any other value
+than `"true"` will leave the feature disabled. By default, the feature will
+also be disabled.
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-legacy-vm
+  annotations:
+    kubevirt.io/legacyWindowsCompat: "true"
+spec:
+  runStrategy: Always
+  template:
+    spec:
+      domain:
+        devices: { }
+        memory:
+          guest: 512Mi
+      terminationGracePeriodSeconds: 180
+      volumes:
+        - containerDisk:
+            image: my.registry/images/legacy-os
+          name: legacy-os
+```
+
+## Alternatives
+
+<!--
+Outline any alternative designs that have been considered
+-->
+
+### Adding a specific field to the core API
+
+The feature could be configured by adding a specific field
+`spec.domain.devices.disablePCIHole64` of type `boolean` to the core API.
+
+Instead of enabling a broader compatibility behavior, this would express the
+configuration change to a `VirtualMachine` exactly.
+
+One advantage of this approach would be the improved discoverability of the
+feature, as it would become part of the core API and its documentation.
+
+The disadvantage of this approach, however, is that users must explicitly know
```

Comment:
As you suggested above, I'd argue that the configurable is more discoverable than an annotation and especially so if we then wrap it in a preference.

Reply:
Right, but flexibility of the API looks to be more important at this point, therefore going to keep the annotation.